core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -116,49 +114,59 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    */
   public static class SSHActionConfig extends PluginConfig {
     @Description("Command to be executed on the remote host. Should include filepath of script and any arguments")
+    @Macro
     private String command;
 
     @Description("Host name of the remote machine where the command needs to be executed.")
+    @Macro
     private String host;
 
     @Nullable
     @Description("Port to connect to. Defaults to 22")
+    @Macro
     private Integer port;
 
     @Description("User name used to connect to host")
+    @Macro
     private String user;
 
-    @Description("File path to Private key")
-    private String privateKeyFile;
+    @Description("The private key to be used to perform the secure shell action. Users can also specify a macro that " +
+      "will pull the private key from the secure key management store in CDAP such as ${secure(myPrivateKey)}.")
+    @Macro
+    private String privateKey;
 
     @Nullable
-    @Description("Password associated with private key")
-    private String password;
+    @Description("Passphrase used to decrypt the provided private key associated with \"privateKeyLookupKey\"")
+    @Macro
+    private String passphrase;
 
     @Nullable
-    @Description("Variable used to hold desired key name to be associated with script output in the instance " +
-      "the script output contains information needed to be stored in the ActionContext for later use in the pipeline.")
-    private String output;
+    @Description("The key used to store the output of the command run by the action. Plugins that run at later " +
+      "stages in the pipeline can retrieve the command's output using this key through macro substitution " +
+      "${outputKey} where \"outputKey\" is the key specified. Defaults to \"sshOutput\".")
+    @Macro
+    private String outputKey;
 
     public SSHActionConfig() {
       port = 22;
+      outputKey = "sshOutput";
     }
 
     @VisibleForTesting
-    public SSHActionConfig(String command, String host, String user, String privateKeyFile, int port, String password,
-                           String output) {
+    public SSHActionConfig(String command, String host, String user, String privateKeyFileLookupKey, Integer port,
+                           String passphrase, String outputKey) {
       this.command = command;
       this.host = host;
-      this.port = port;
+      this.port = (port == null) ? 22 : port;
       this.user = user;
-      this.privateKeyFile = privateKeyFile;
-      this.password = password;
-      this.output = output;
+      this.privateKey = privateKeyFileLookupKey;
+      this.passphrase = passphrase;
+      this.outputKey = outputKey;
     }
 
     public void validate() {
       //check that port is not negative
-      if (port < 0) {
+      if (!containsMacro("port") && port < 0) {
```

Comment:
Oh, I thought that we went with an empty set at runtime. Just checked PluginInstantiator and that's not the case.. will fix.